### PR TITLE
10.x.x/fix/terrain opacity blend

### PR DESCRIPTION
**Pull request by spaceburp**
### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [x] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [x] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?
Backport for case 1283124. Parent PR https://github.com/Unity-Technologies/Graphics/pull/3083

Fogbugz Link: https://fogbugz.unity3d.com/f/cases/1286881/

---
### Testing status
Describe what manual/automated tests were performed for this PR
- Manually tested painting and verifying that artifacts previously seen have been removed
- Manually tested 2 - 4 layer blending with different layers with different Opacity as Density states
- Manually tested +4 layer blending and verified that density blending is disabled with >4 Terrain Layers
- Manually tested Heightbased Blending feature for < 4 layers to make sure nothing broke there
- Ran automated tests, specifically the one that was originally failing and made us aware of this issue in the first place: 038_Terrain_Blend
- Created automated test for Opacity as Density blend with < 4 layers

Before changes:

After changes:

4 Layers with 1 layer using Opacity As Density blend. Ignore the black pixels that show. This is because the splat is fully weighted towards that Terrain Layer so it is pulling the black from the Diffuse Map and is not a blending error

https://user-images.githubusercontent.com/1635624/107831371-c7a9df80-6d42-11eb-83cf-c54a4483efc9.mp4

Example of Opacity as Density blending being disabled if there is more than 4 Terrain Layers on the Terrain

https://user-images.githubusercontent.com/1635624/107831379-ca0c3980-6d42-11eb-816c-4b840462852f.mp4

Example of 4 Layers with 2 layers using Opacity As Density Blend

https://user-images.githubusercontent.com/1635624/107831381-cb3d6680-6d42-11eb-9ea3-e90401cf7a68.mp4

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.universal/CHANGELOG.md [hunk 1]
```diff
@@ -6,8 +6,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [10.4.0] - 2020-01-26
 
-Version Updated
-The version number for this package has increased due to a version update of a related graphics package.
+### Changed
```
Comment (spaztronaut) [inline com.unity.render-pipelines.universal/CHANGELOG.md:9]: is this the correct spot for this changelog modification? under 10.4.0

Comment (phi-lira) [reply com.unity.render-pipelines.universal/CHANGELOG.md:9]: Yes

**Review by phi-lira — APPROVED** [review]

**Review by GrantLamb-Unity — APPROVED** [review]
Thanks for providing your test artifacts for these changes!

**Comment by wyatttttt** [comment]
failing URP tests on this branch:

### Build_Universal_iPhone_Metal_Standalone_cache_il2cpp_Linear_2020.2
- UnhandledLogMessageException: Unhandled log message: '[Assert] Converting invalid MinMaxAABB'. Use UnityEngine.TestTools.LogAssert.Expect
- Run Failed: One or more errors in a prebuild setup. See the editor log for details.

### Build Universal on Android_Vulkan_il2cpp_Linear_Standalone_cache_build_Player on version 2020.2
- UnhandledLogMessageException: Unhandled log message: '[Assert] Converting invalid MinMaxAABB'. Use UnityEngine.TestTools.LogAssert.Expect
- Run Failed: One or more errors in a prebuild setup. See the editor log for details.
- Assertion failed on expression: 'm_LastAlloc == NULL'

### Build Universal on Android_OpenGLES3_il2cpp_Linear_Standalone_cache_build_Player on version 2020.2
- UnhandledLogMessageException: Unhandled log message: '[Assert] Converting invalid MinMaxAABB'. Use UnityEngine.TestTools.LogAssert.Expect
- Run Failed: One or more errors in a prebuild setup. See the editor log for details.
- Assertion failed on expression: 'm_LastAlloc == NULL'

### Build_Universal_Win_DX12_Standalone_cache_mono_Linear_2020.2
- UnhandledLogMessageException: Unhandled log message: '[Assert] Converting invalid MinMaxAABB'. Use UnityEngine.TestTools.LogAssert.Expect
- Run Failed: One or more errors in a prebuild setup. See the editor log for details.
- Assertion failed on expression: 'm_LastAlloc == NULL'

### Build Universal on Win_DX11_mono_Linear_Standalone_cache_build_Player on version 2020.2
- UnhandledLogMessageException: Unhandled log message: '[Assert] Converting invalid MinMaxAABB'. Use UnityEngine.TestTools.LogAssert.Expect
- Run Failed: One or more errors in a prebuild setup. See the editor log for details.
- Assertion failed on expression: 'm_LastAlloc == NULL'

### Universal on OSX_Metal_playmode_cache_mono_Linear on version 2020.2
- UnhandledLogMessageException: Unhandled log message: '[Assert] Converting invalid MinMaxAABB'. Use UnityEngine.TestTools.LogAssert.Expect
- Run Failed: One or more errors in a prebuild setup. See the editor log for details.

### Universal on Win_DX11_playmode_XR_cache_mono_Linear on version 2020.2
- UnhandledLogMessageException: Unhandled log message: '[Assert] Converting invalid MinMaxAABB'. Use UnityEngine.TestTools.LogAssert.Expect
- Run Failed: One or more errors in a prebuild setup. See the editor log for details.
- Assertion failed on expression: 'm_LastAlloc == NULL'

**Comment by wyatttttt** [comment]
looks like iphone terrain is not rendering

**Review by Verasl — APPROVED** [review]
LGTM

**Review by lumonix — APPROVED** [review]
Approved to disable this test for iphone like the test it copied (was also disabled for iphone)

**Review by oleks-k — APPROVED** [review]
Looks good!

**Pull request merged**